config/sailfish-qdoc-template.prf

```diff
@@ -0,0 +1,34 @@
+QDOC = $$[QT_INSTALL_BINS]/qdoc
+QHELPGENERATOR = $$[QT_INSTALL_BINS]/qhelpgenerator
+
+# Note: not $${QT_INSTALL_DOCS}/sailfish-qdoc-template
+# because the SDK prevents installation to /usr/share/doc/
+isEmpty(SAILFISH_QDOC.template):SAILFISH_QDOC.template=/usr/share/sailfish-qdoc-template
+isEmpty(SAILFISH_QDOC.style):SAILFISH_QDOC.style=offline
+
+QDOCCONF = $${SAILFISH_QDOC.template}/$${SAILFISH_QDOC.style}.qdocconf
+QHELPFILE = html/$${SAILFISH_QDOC.project}.qhp
+QCHFILE = $${SAILFISH_QDOC.project}.qch
+
+ENV += \
+    QT_VERSION=$${QT_MAJOR_VERSION}.$${QT_MINOR_VERSION} \
+    PROJECT_CONFIG=$${_PRO_FILE_PWD_}/$${SAILFISH_QDOC.config} \
+    BUILD_DIRECTORY=$${OUT_PWD} \
+    TEMPLATE=$${SAILFISH_QDOC.template}
+
+docs.input = ..
+docs.output = $${QCHFILE}
+docs.commands = (env $$ENV $$QDOC $$QDOCCONF) && \
+                    ($${QHELPGENERATOR} $${QHELPFILE} -o $${QCHFILE})
+docs.CONFIG += no_check_exist no_default_install
+
+docs_install.files = $${QCHFILE}
+docs_install.path = $${SAILFISH_QDOC.path}
+docs_install.CONFIG += no_check_exist no_build
+
+QMAKE_EXTRA_TARGETS += docs
+PRE_TARGETDEPS += docs
+
+INSTALLS += docs_install
```

Comment:
I would prefer not implying those. It's common that docs building doesn't come into the way of regular edit-build-test cycle, i.e., that the docs+install_docs targets need to be required explicitly.

Reply:
Disagree. Qdoc processes the documentation we have nearly instantly and consider it unlikely that many would be actually manually modifying the build system to save a second or two. If they want, it's likely still pretty easy by commenting out the doc subdir in the projects or something like that.

In the general case don't see why the documentation should get any special treatment from the other sub-targets. We don't do make translations & make translations_install, or plugins etc. And the point of this kind of templates is to remove repetitive content.

If this starts to matter in some project, it should be easy later on to add e.g. a different config option as a variant that requires explicit build & install. Or the projects can introduce their own CONFIG whether the doc subdirectory gets included in the build.